Purchases/Public/RCPurchases.m

```diff
@@ -673,12 +679,30 @@ - (void)checkTrialOrIntroductoryPriceEligibility:(NSArray<NSString *> *)productI
                                  completionBlock:(RCReceiveIntroEligibilityBlock)receiveEligibility
 {
     [self receiptData:^(NSData * _Nonnull data) {
-        [self.backend getIntroEligibilityForAppUserID:self.appUserID
-                                          receiptData:data
-                                   productIdentifiers:productIdentifiers
-                                           completion:^(NSDictionary<NSString *,RCIntroEligibility *> * _Nonnull result) {
-                                               CALL_IF_SET_ON_MAIN_THREAD(receiveEligibility, result);
-                                           }];
+        LocalReceiptParser *receiptParser = [[LocalReceiptParser alloc] init];
+        [receiptParser checkTrialOrIntroductoryPriceEligibilityWithData:data
+                                                     productIdentifiers:productIdentifiers
+                                                             completion:^(NSDictionary<NSString *, NSNumber *> * _Nonnull receivedElegilibity,
+                                                                          NSError * _Nullable error) {
+            if (!error) {
+                NSMutableDictionary<NSString *, RCIntroEligibility *> *convertedEligibility = [[NSMutableDictionary alloc] init];
+                
+                for (NSString *key in receivedElegilibity.allKeys) {
+                    convertedEligibility[key] = [[RCIntroEligibility alloc] initWithEligibilityStatusCode:receivedElegilibity[key].intValue];
+                }
```

Comment:
I'm doing the conversion here, which is the one byproduct of not using the same class. I would move this somewhere else before finishing, though.